.github/workflows/gosec.yaml

```diff
@@ -22,6 +22,11 @@ jobs:
       - name: Run Gosec Security Scanner
         uses: securego/gosec@master
         with:
-          args: -exclude-generated ./...
+          # we let the report trigger a failure using the GitHub Security features.
+          args: -exclude-generated -no-fail -fmt sarif -out results.sarif ./...
```

Comment:
They have the `-no-fail` in [the documentation](https://github.com/securego/gosec#integrating-with-code-scanning), but I'm not completely sure about this. I think it would make sense to still fail here even with the warnings/errors shown in the UI. I wonders what others think.

Reply:
I think we should still fail

Reply:
OK, thanks for the input. I tried removing the `-no-fail`. But the problem now is that if the `Run Gosec Security Scanner` step fails, the rest of the job is skipped (including the `Upload SARIF file` step). So the warning in the UI will not show up.

Reply:
https://github.com/redhat-appstudio/release-service/actions/runs/5728029661/job/15521693572?pr=228

```
Run if [ "failure" != "success" ]
Gosec failed. The errors and warnings should be displayed in the PR.
Error: Process completed with exit code 1.
```